docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,262 @@
+type ObjectFieldSelector struct {
+        // Required: Version of the schema the FieldPath is written in terms of.
+        // If no value is specified, it will be defaulted to the APIVersion of the
+        // enclosing object.
+        APIVersion string `json:"apiVersion"`
+        // Required: Path of the field to select in the specified API version
+        FieldPath string `json:"fieldPath"`
+}
+```
+
+#### Examples
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh", "-c", "env" ]
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            fieldRef:
+              fieldPath: .spec.containers[?(@.name=="container-name")].resources.limits.cpu
+  restartPolicy: Never
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox
+      command: ["sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi; sleep 5; done"]
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            fieldRef:
+              fieldPath: .spec.containers[?(@.name=="container-name")].resources.limits.cpu
+```
+
+### API with partial json path selectors
+
+Partial json path selectors specify paths to resources limits and requests relative to the container spec.
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and limits to be exposed as environment variables.
+
+| Name | Selectors |
+| -------------------- | ------------------- |
+| CPU_LIMIT | resources.limits.cpu |
+| MEMORY_LIMIT | resources.limits.memory |
+| STORAGE_LIMIT | resources.limits.storage |
+| CPU_REQUEST | resources.requests.cpu |
+| MEMORY_REQUEST | resources.requests.memory |
+| STORAGE_REQUEST | resources.requests.storage |
+
+Since envionment variables are container scoped, so there is no need to specify container name as part of the partial selectors as they are relative to container spec.
+
+#### Volume plugin
+
+| Path | Selectors |
+| -------------------- | ------------------- |
+| container_name/cpu_limit | resources.limits.cpu |
+| container_name/memory_limit | resources.limits.memory |
+| container_name/storage_limit | resources.limits.storage |
+| container_name/cpu_request | resources.requests.cpu |
+| container_name/memory_request | resources.requests.memory |
+| container_name/storage_request | resources.requests.memory |
+
+Since envionment variables are container scoped, so container name must be specified as part of their path name. The format is that container name and path name must be separated by slash (`/`).
```

Comment:
@thockin thanks for your comment. Could you please elaborate on "not safe" part?